modules/tinymce/src/plugins/emoticons/main/ts/core/EmojiDatabase.ts

```diff
@@ -98,23 +98,27 @@ const initDatabase = (editor: Editor, databaseUrl: string, databaseId: string):
     }, (err) => {
       // eslint-disable-next-line no-console
       console.log(`Failed to load emoticons: ${err}`);
-      categories.set(Optional.some({}));
-      all.set(Optional.some([]));
+      categories.set(Maybes.just({}));
+      all.set(Maybes.just([]));
     });
   });
 
   const listCategory = (category: string): EmojiEntry[] => {
     if (category === ALL_CATEGORY) {
       return listAll();
     }
-    return categories.get().bind((cats) => Optional.from(cats[category])).getOr([]);
+    return Fun.pipe(
+      categories.get(),
+      Maybes.bindNullable((cats) => cats[category]),
+      Maybes.getOr([])
+    );
   };
 
-  const listAll = (): EmojiEntry[] => all.get().getOr([]);
+  const listAll = (): EmojiEntry[] => Maybes.getOr<EmojiEntry[]>([])(all.get());
```

Comment:
Should this be using `Fun.pipe`? The use of it in this file is rather inconsistent which I'm not a fan of.

Reply:
Could we have curried and uncurried versions. Sometimes you might just want to  call a single function going though a pipe just to avoid the uncurry seems a bit strange. Maybe have things like Maybes.mapU that is uncurried.